### PR TITLE
DAG edge remover and detector

### PR DESCRIPTION
**Pull request by maddy-tod**
### Summary
This adds the enhancements raised in #1501. It wraps the multigraph functions to detect/remove edges into more convenient functions on the DAG. This allows the user to select a specific edge when there are multiple edges between 2 nodes. The functions take the same format as the `dag.multigraph` functions, but take an additional optional parameter `wire` which is used to identify which wire the user means. The wire is of the form `(QuantumRegister(size, name), index)`.

### CODE REVIEW

File: qiskit/dagcircuit/_dagcircuit.py [hunk 1]
```diff
@@ -1145,6 +1145,42 @@ def remove_nondescendants_of(self, node):
             if nd["type"] == "op":
                 self._remove_op_node(n)
 
+    def remove_edge(self, node1, node2, wire=None):
```
Comment (kdk) [inline qiskit/dagcircuit/_dagcircuit.py]: It doesn't look like `wire` is optional here. (We'll always raise if `wire` is `None`). In that case, we should make it a required argument and update the docstring.

File: test/python/test_dagcircuit.py [hunk 2]
```diff
@@ -209,6 +209,34 @@ def test_nodes_in_topological_order(self):
...
                          [i for i in named_nodes])
 
+    def test_dag_has_edge(self):
+        """ Test that existence of edges between nodes is correctly identified"""
+        self.assertTrue(self.dag.has_edge(1, 2))
+        self.assertTrue(self.dag.has_edge(1, 2, (QuantumRegister(3, 'qr'), 0)))
+
+        self.assertFalse(self.dag.has_edge(1, 3))
+        self.assertFalse(self.dag.has_edge(1, 3, (QuantumRegister(3, 'qr'), 0)))
+
```
Comment (kdk) [inline test/python/test_dagcircuit.py]: Can you add a test case checking that we return false when an edge exists but doesn't match the specified wire?

File: qiskit/dagcircuit/_dagcircuit.py (hunk 1 above)
Comment (maddy-tod) [reply qiskit/dagcircuit/_dagcircuit.py]: I have updated the code so that wire is optional - it removes a random edge between the two given nodes.

File: test/python/test_dagcircuit.py (hunk 2 above)
Comment (maddy-tod) [reply test/python/test_dagcircuit.py]: Sure, I have just added one
